spark/src/main/scala/org/apache/comet/serde/QueryPlanSerde.scala

```diff
@@ -2718,11 +2718,6 @@ object QueryPlanSerde extends Logging with ShimQueryPlanSerde with CometExprShim
           return None
         }
 
-        if ((join.leftKeys ++ join.rightKeys).exists(_.dataType.isInstanceOf[StructType])) {
-          withInfo(join, "Unsupported struct data type in join keys")
-          return None
-        }
```

Comment:
We should further check if the data types are supported using `supportedDataType`.

Reply:
So instead of removing this condition we change it to something like
```scala
        if ((join.leftKeys ++ join.rightKeys).exists(e => !supportedDataType(e.dataType, allowStruct = true))) {
          withInfo(join, "Unsupported data type in join keys")
          return None
        }
```
?

Reply:
Oh, I rethink about this. `supportedDataType` doesn't mean Join operator key support. We don't need it.

Reply:
Removed.